flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NettyMessage.java

```diff
@@ -259,7 +259,7 @@ protected Object decode(ChannelHandlerContext ctx, ByteBuf in) throws Exception
 		static final byte ID = 0;
 
 		// receiver ID (16), sequence number (4), backlog (4), dataType (1), isCompressed (1), buffer size (4)
-		static final int MESSAGE_HEADER_LENGTH = 16 + 4 + 4 + 1 + 1 + 4;
+		static final int MESSAGE_HEADER_LENGTH = InputChannelID.getByteBufLength() + Integer.BYTES + Integer.BYTES + Byte.BYTES + 1 + Integer.BYTES;
```

Comment:
why `isCompressed` (boolean) is hardcoded, while everything else is not?

Reply:
After a deeper investigation, I find that a `boolean` will be treated as a `byte` in netty. Will change it to `Byte.BYTES`.